python/cudf/cudf/_lib/parquet.pyx

```diff
@@ -301,6 +290,69 @@ cpdef write_parquet(
     else:
         return None
 
+
+cdef class ParquetChunkedWriter:
+    """
+    ParquetChunkedWriter lets you write out a series of Tables to a single Parquet file
+
+    See Also
+    --------
+    cudf.io.parquet.write_parquet
+    """
+    cdef shared_ptr[pq_chunked_state] state
+    cdef cudf_io_types.sink_info sink
+    cdef cudf_io_types.statistics_freq stat_freq
+    cdef cudf_io_types.compression_type comp_type
+    cdef object index
+
+    def __cinit__(self, path, index=None, compression=None, statistics="ROWGROUP"):
+        cdef string filepath = <string>str(path).encode()
+        self.sink = cudf_io_types.sink_info(filepath)
+        self.stat_freq = _get_stat_freq(statistics)
+        self.comp_type = _get_comp_type(compression)
+        self.index = index
+
+    def write(self, Table table):
+        """ Writes a single table to the file """
+        if not self.state:
+            self._initialize_chunked_state(table)
+
+        cdef table_view tv = table.data_view()
+        if self.index is not False:
+            if isinstance(table._index, cudf.core.multiindex.MultiIndex) \
+                    or table._index.name is not None:
+                tv = table.view()
+
+        with nogil:
+            write_parquet_chunked(tv, self.state)
+
+    def close(self):
+        if self.state:
+            with nogil:
+                write_parquet_chunked_end(self.state)
+
+    def __del__(self):
+        self.close()
```

Comment:
`__del__` isn't guaranteed to run where it's not a safe to put a cleanup method into it. We could use `__dealloc__` but that requires not needing the GIL, but ideally this behavior could just be controlled via a smart pointer instead.

Reply:
Changed to use __dealloc__ - we don't need the GIL to run this cleanup code